doc/developer/sqllogictest.md

```diff
@@ -0,0 +1,222 @@
+    * the result looks like this in `mode standard`:
+        ```
+        query TT colnames
+        select col1 col2 from atable
+        ----
+        col1 col2
+        row1col1
+        row1col2
+        row2col1
+        row2col2
+        ```
+    * and in `mode cockroach`:
+        ```
+        query TT colnames
+        select col1 col2 from atable
+        ----
+        col1 col2
+        row1col1 row1col2
+        row2col1 row2col2
+        ```
+
+    In `mode cockroach`, if your expected column name or row entry contains a space, to have the space not treated as a column delimiter, replace the space with the symbol ␠, which is U+2040, the Unicode "SYMBOL FOR SPACE".
+    Example:
+    ```
+    query TT colnames
+    select "col 1" "col 2" from atable
+    ----
+    col␠1 col␠2
+    row1␠col1 row1␠col2
+    row2␠col1 row2␠col2
+    ```
+
+    As mentioned earlier, in `mode cockroach`, replacing the space with the symbol ␠ is optional when there is only one column of expected output. This will work fine:
+    ```
+    query T colnames
+    select "col 1" from atable
+    ----
+    col 1
+    row1 col1
+    row2 col1
+    ```
+
+    In `mode standard`, it does not seem to possible to indicate that the expected column name contains a space even if only one column of output is expected.
+
+2. In `mode standard`, a Decimal in the query result is treated as having a scale of 4 if its actual scale is 0 and as having a scale of 3 otherwise. A floating point number is always treated as having a scale of 3.
+    ```
+    mode standard
+
+    query R
+    SELECT 384::decimal(38, 5)
+    ----
+    384.000
+
+    query R
+    SELECT 1::decimal(38, 0)
+    ----
+    1.0000
+
+    query R
+    SELECT 2::float
+    ----
+    2.000
+
+    query R
+    SELECT 3.143892789739843::float
+    ----
+    3.144
+
+    query R
+    SELECT 314.3820897445435::float
+    ----
+    314.382
+    ```
+
+    `mode cockroach`, on the other hand, respects the scale of a Decimal type and the precision of a floating point number.
+
+    ```
+    mode cockroach
+
+    query R
+    SELECT 384::decimal(38, 5)
+    ----
+    384.00000
+
+    query R
+    SELECT 1::decimal(38, 0)
+    ----
+    1
+
+    query R
+    SELECT 2::float
+    ----
+    2
+
+    query R
+    SELECT 3.143892789739843::float
+    ----
+    3.143892789739843
+
+    query R
+    SELECT 314.3820897445435::float
+    ----
+    314.3820897445435
+    ```
+
+### Boolean support
+
+SQLite tests interpret booleans as integers with value 1 or 0. CockroachDB tests interpret booleans as booleans with values true or false. Our sqllogictest supports treating booleans as either integers or booleans for both modes. The tests below will pass in both `mode standard` and `mode cockroach`.
```

Comment:
There's a bit more intricacy here, actually. sqllogictest as designed by SQLite shipped two coercions for booleans, integers (`I`) and text (`T`), where the conversions are as you would expect: 0 and 1 for integers, and false and true for text. Whoever implemented sqllogictest at Cockroach failed to understand this, and added an additional format code (`B`), which is equivalent to text in every way—except that I guess it will error if you have a string "true" in a column?

New code should not use this unnecessary `B` format code, and should just use `T`. (`I` is fine, I guess; it's just a less clear version of `T`.)

A tenant of sqllogictest: sqllogictest is not for checking the types of the results. That's probably worth calling out somewhere. If you are attempting to write a test to verify that you are getting the right result type, sqllogictest is _not_ the place for that, as it will implicitly coerce things all over the place. Those sorts of tests belong as integration tests in Rust code.

Reply:
We don't seem to actually support using T for booleans. I ran this:
```
query T
select true
----
true
```

And I got:
```
thread 'main' panicked at 'Don't know how to format (Text, True)', src/sqllogictest/runner.rs:356:34
```

I can add a coercion from boolean to text if you like.

Reply:
Oh, d'oh, yeah, I guess we're not compatible with sqllite's runner in that way. We should probably add that cast, but it's not particularly important.